runtime/compilers/rillv1/parse_rillyaml.go

```diff
@@ -92,10 +95,20 @@ func (p *Parser) parseRillYAML(ctx context.Context, path string) error {
 	}
 
 	tmp := &rillYAML{}
+
 	if err := yaml.Unmarshal([]byte(data), tmp); err != nil {
 		return newYAMLError(err)
 	}
 
+	dec := yaml.NewDecoder(strings.NewReader(data))
+	dec.KnownFields(true)
+
+	err = dec.Decode(tmp)
+
+	if err != nil && !errors.Is(err, io.EOF) {
```

Comment:
nit: Go convention is to handle an `err` in the same code block as the one where it's set
```suggestion
	dec := yaml.NewDecoder(strings.NewReader(data))
	dec.KnownFields(true)

	err = dec.Decode(tmp)
	if err != nil && !errors.Is(err, io.EOF) {
```